src/control/cmd/dmg/system.go

```diff
@@ -239,3 +240,40 @@ func (cmd *systemStartCmd) Execute(_ []string) (errOut error) {
 
 	return resp.Errors()
 }
+
+type systemCleanupCmd struct {
+	logCmd
+	cfgCmd
+	ctlInvokerCmd
+	jsonOutputCmd
+	Machine string `long:"machine" short:"m" description:"Machine to cleanup"`
+	Verbose bool   `bool:"verbose" short:"v" description:"Output additional cleanup information"`
+}
+
+func (cmd *systemCleanupCmd) Execute(_ []string) (errOut error) {
+	defer func() {
+		errOut = errors.Wrap(errOut, "system cleanup failed")
+	}()
+
+	if cmd.config == nil {
+		return errors.New("no configuration loaded")
+	}
+
+	ctx := context.Background()
+	req := new(control.SystemCleanupReq)
+	req.SetSystem(cmd.config.SystemName)
+	req.Machine = cmd.Machine
+
+	resp, err := control.SystemCleanup(ctx, cmd.ctlInvoker, req)
+	if err != nil {
+		return err // control api returned an error, disregard response
+	}
+
+	if cmd.jsonOutputEnabled() {
+		return cmd.outputJSON(resp, err)
+	}
+
+	cmd.log.Infof("Response from SystemCleanup RPC: %+v", resp)
+
+	return nil
```

Comment:
we have adopted the habit of returning resp.Errors() where possible so we have dmg exit nonzero when there are rank failures in the results nested inside the response.

Reply:
I can see that there are not any rank results in the response but it might be prudent to change the return as above anyway in case someone modifies Errors() on the response in the future (and then is confused as to why dmg is exiting with zero with response errors).

Reply:
Ok fixed. I don't have any errors which would be indicated by the response. We either fail the whole call or pass the whole call so my Errors() implementation still returns nil for now.